packages/mml-web/src/camera/DragFlyCameraControls.ts

```diff
@@ -203,4 +230,156 @@ export class DragFlyCameraControls {
     // restrict to a reasonable min and max
     this.speed = Math.max(5, Math.min(this.speed, 1000));
   }
+
+  private handleTouchStart(event: TouchEvent) {
+    event.preventDefault();
+    let startX: number;
+    let startY: number;
+
+    for (const touch of Array.from(event.touches)) {
+      if (!this.touchesMap.has(touch.identifier)) {
+        startX = touch.clientX;
+        startY = touch.clientY;
+
+        this.touchesMap.set(touch.identifier, {
+          touch,
+          startX,
+          startY,
+          currentX: startX,
+          currentY: startY,
+        });
+      }
+    }
+
+    if (event.touches.length === 1) {
+      this.panStartX = event.touches[0].clientX;
+      this.panStartY = event.touches[0].clientY;
+      this.clickTimestamp = Date.now();
+    }
+  }
+
+  private handleTouchEnd(event: TouchEvent) {
+    if (this.isMoving) {
+      this.zoomTimestamp = Date.now();
+    }
+    this.isMoving = false;
+
+    const remainingTouches = new Set(Array.from(event.touches).map((touch) => touch.identifier));
+
+    for (const [touchId] of this.touchesMap) {
+      if (!remainingTouches.has(touchId)) {
+        this.touchesMap.delete(touchId);
+      }
+    }
+
+    if (Date.now() - this.clickTimestamp < this.clickThresholdMilliseconds) {
+      /* this is a click */
+      // Create and dispatch a new mouse event with specific x and y coordinates
+      const clickEvent = new MouseEvent("click", {
```

Comment:
This seems extraneous now as the `MMLClickTrigger` handles clicks without this logic.